packages/react-components/src/components/kpi/tests/kpi.spec.tsx

```diff
@@ -0,0 +1,226 @@
+        query={ERROR_QUERY}
+        settings={{
+          showAggregationAndResolution: false,
+          showAssetName: false,
+          showDataQuality: false,
+          showName: false,
+          showTimestamp: false,
+          showUnit: false,
+        }}
+      />
+    );
+
+    // KPI no values should be visible, only the error component
+    expect(screen.queryByTestId('kpi-name-and-unit')).toBeNull();
+    expect(screen.queryByTestId('kpi-timestamp')).toBeNull();
+    expect(screen.queryByTestId('kpi-timestamp')).toBeNull();
+    expect(screen.queryByTestId('kpi-aggregation')).toBeNull();
+    expect(screen.queryByTestId('kpi-value')).toBeNull();
+
+    expect(screen.getByTestId('kpi-error-component').textContent).toContain(
+      ERROR_DATA_STREAM.error.msg
+    );
+  });
+
+  it('kpi widget default decimal places', () => {
+    render(<KPI query={SUCCESS_QUERY} />);
+
+    // KPI value should be visible with the defaut 4 decimal places
+    expect(screen.getByTestId('kpi-value').textContent).toContain(
+      `${DATA_STREAM.data[0].y.toFixed(4)}`
+    );
+  });
+
+  it('kpi widget custom decimal places', () => {
+    render(<KPI query={SUCCESS_QUERY} significantDigits={5} />);
+
+    // KPI value should be visible with 5 decimal places
+    expect(screen.getByTestId('kpi-value').textContent).toContain(
+      `${DATA_STREAM.data[0].y.toFixed(5)}`
+    );
+  });
+
+  it('kpi widget no decimal places', () => {
+    render(<KPI query={SUCCESS_QUERY} significantDigits={0} />);
+
+    // KPI value should be visible with no decimal places
+    expect(screen.getByTestId('kpi-value').textContent).toContain(
+      `${DATA_STREAM.data[0].y.toFixed(0)}`
+    );
+  });
+
+  it('kpi widget with custom green background color', () => {
+    render(
+      <KPI query={SUCCESS_QUERY} settings={{ backgroundColor: '#00ff00' }} />
+    );
+
+    // background color is set
+    expect(screen.getByTestId('kpi-base-component')).toHaveStyle(
+      'background-color: #00ff00'
+    );
+
+    // font color is calculated to use the highest contrast color (in this case it is black)
+    expect(screen.getByTestId('kpi-value')).toHaveStyle('color: black');
+  });
+
+  it('kpi widget with custom green background color and filled threshold', () => {
+    render(
+      <KPI
+        query={SUCCESS_QUERY}
+        settings={{ backgroundColor: '#00ff00' }}
+        thresholds={[FILLED_THRESHOLD]}
+      />
+    );
+
+    // background color is set to the threshold color and NOT the custom background color
+    expect(screen.getByTestId('kpi-base-component')).toHaveStyle(
+      'background-color: #ff0000'
+    );
+  });
+
+  it('kpi widget in error state with custom green background color', () => {
+    render(
+      <KPI query={ERROR_QUERY} settings={{ backgroundColor: '#00ff00' }} />
+    );
+
+    // background color is set to the custom background color
+    expect(screen.getByTestId('kpi-error-component')).toHaveStyle(
+      'background-color: #00ff00'
```

Comment:
Create variables for any values you setup at the start of a test which you are writing assertions for.